stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -307,7 +307,69 @@ static void _addImageProtocolConformances(const mach_header *mh,
   
   _addImageProtocolConformancesBlock(conformances, conformancesSize);
 }
+
+#elif defined(__ANDROID__)
+#include <climits>
+#include <cstdio>
+#include <unordered_set>
+#include <unistd.h>
+
+static void _addImageProtocolConformances(const char *name) {
+  void *handle = dlopen(name, RTLD_LAZY);
+  if (!handle)
+    return; // not a shared library
+  auto conformances = reinterpret_cast<const uint8_t*>(
+      dlsym(handle, SWIFT_PROTOCOL_CONFORMANCES_SECTION));
+
+  if (!conformances) {
+    // if there are no conformances, don't hold this handle open.
+    dlclose(handle);
+    return;
+  }
+
+  // Extract the size of the conformances block from the head of the section
+  auto conformancesSize = *reinterpret_cast<const uint64_t*>(conformances);
+  conformances += sizeof(conformancesSize);
+
+  _addImageProtocolConformancesBlock(conformances, conformancesSize);
+
+  dlclose(handle);
+}
+
+static void android_iterate_libs(void (*callback)(const char*)) {
```

Comment:
Just I made #1466, which implemented the idea I mentioned you several days ago in #1108 (Cygwin).
It is factored only for Linux/Cygwin (except _APPLE_), but I think many lines could be shared with Android.